### ISSUE TITLE
golang plugin doesn't work as documentation says

### ISSUE THREAD

**Issue by ana** [avocado-framework/avocado#4995]
golang plugin doesn't work as the documentation at https://avocado-framework.readthedocs.io/en/91.0/plugins/optional/golang.html says. Specifically, I haven't managed to get `avocado list` listing any GOLANG tests, there is some step missing there that isn't obvious or this is because changes on `go test` and/or `nrunner`.

@clebergnu Could you please check if it works for you (git log says you are the one who worked on this) and tell me quickly the steps you followed if it works? I'll be able to update the documentation then, thanks!

**Comment by clebergnu** [comment]
OK, turns out this issue is more on the GO side, than on Avocado's side.  But, we can at the very least improve the documentation.

First, I believe we should change the documentation to use the example GO tests we ship, so that it won't depend on extra GO packages (AFAICT, the base installation won't include the test files referenced in the current documentation).  To do that, we need to tweak the [GOPATH](https://golang.org/doc/install/source#environment) variable according (there's no way around it).  This means:

```
cd $AVOCADO_SOURCE_ROOT
export GOPATH=`pwd`/optional_plugins/golang/tests
```

With that, one should be able to see the `countavocado` tests:

```
$ avocado -V list countavocados
Type   Test                              Tag(s)
golang countavocados:TestEmptyContainers
golang countavocados:TestNoContainers
golang countavocados:ExampleContainers

Resolver             Reference     Info
avocado-instrumented countavocados File "countavocados" does not end with ".py"
exec-test            countavocados File "countavocados" does not exist or is not a executable file

TEST TYPES SUMMARY
==================
golang: 3
```

But, while the Avocado test resolution does not involve running `go`, running it does.  On first try, you'd get:

```
avocado run countavocados
JOB ID     : f167e92b3fbffb481ab6e74cda2e8589dace9224
JOB LOG    : /home/cleber/avocado/job-results/job-2021-09-30T11.47-f167e92/job.log
 (1/3) countavocados:TestEmptyContainers: STARTED
 (2/3) countavocados:TestNoContainers: STARTED
 (3/3) countavocados:ExampleContainers: STARTED
 (1/3) countavocados:TestEmptyContainers: FAIL (0.50 s)
 (2/3) countavocados:TestNoContainers: FAIL (0.50 s)
 (3/3) countavocados:ExampleContainers: FAIL (0.50 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 3 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2021-09-30T11.47-f167e92/results.html
JOB TIME   : 1.84 s
```

With the test failures being all something like:

```
[stderr] package countavocados is not in GOROOT (/usr/lib/golang/src/countavocados)
```

Looks like GO 1.11 changed a lot about the `GOPATH` handling (disabling it by default, and requiring a more formal declaration of modules -- called "Module Aware" mode).  But it can be [disabled](https://golang.org/ref/mod#mod-commands) with:

```
go env -w GO111MODULE=off
```

After that, it should find the modules on `GOPATH`:

```
 avocado run countavocados
JOB ID     : 2b0ce45415b6855ebbb66d8fab971d5e316312c1
JOB LOG    : /home/cleber/avocado/job-results/job-2021-09-30T11.49-2b0ce45/job.log
 (1/3) countavocados:TestEmptyContainers: STARTED
 (2/3) countavocados:TestNoContainers: STARTED
 (3/3) countavocados:ExampleContainers: STARTED
 (1/3) countavocados:TestEmptyContainers: PASS (0.50 s)
 (2/3) countavocados:TestNoContainers: PASS (0.50 s)
 (3/3) countavocados:ExampleContainers: PASS (0.50 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2021-09-30T11.49-2b0ce45/results.html
JOB TIME   : 1.93 s
```

**Comment by clebergnu** [comment]
BTW, the other possibility is to require some golang package with tests to be installed, such as `golang-tests` on Fedora.  While that, it should be possible to:

```
GOPATH=/usr/lib/golang avocado run bufio
JOB ID     : 9feb393713703fce6cbf9e4d4fc75bfbb49ae722
JOB LOG    : /home/cleber/avocado/job-results/job-2021-09-30T12.04-9feb393/job.log
 (01/77) bufio:TestReaderSimple: STARTED
 (04/77) bufio:TestReadRune: STARTED
 (05/77) bufio:TestUnreadRune: STARTED
 (02/77) bufio:TestReader: STARTED
 (03/77) bufio:TestZeroReader: STARTED
 (07/77) bufio:TestNoUnreadByteAfterPeek: STARTED
 (06/77) bufio:TestNoUnreadRuneAfterPeek: STARTED
 (08/77) bufio:TestUnreadByte: STARTED
 (01/77) bufio:TestReaderSimple: PASS (0.50 s)
 (04/77) bufio:TestReadRune: PASS (0.51 s)
 (05/77) bufio:TestUnreadRune: PASS (0.51 s)
 (02/77) bufio:TestReader: PASS (0.50 s)
 (03/77) bufio:TestZeroReader: PASS (0.50 s)
 (07/77) bufio:TestNoUnreadByteAfterPeek: PASS (0.51 s)
 (08/77) bufio:TestUnreadByte: PASS (0.51 s)
 (06/77) bufio:TestNoUnreadRuneAfterPeek: PASS (0.52 s)
 (09/77) bufio:TestUnreadByteMultiple: STARTED
 (10/77) bufio:TestUnreadByteOthers: STARTED
 (12/77) bufio:TestUnreadRuneAtEOF: STARTED
 (11/77) bufio:TestUnreadRuneError: STARTED
 (13/77) bufio:TestReadWriteRune: STARTED
 (15/77) bufio:TestWriter: STARTED
 (14/77) bufio:TestReadStringAllocs: STARTED
 (16/77) bufio:TestWriteErrors: STARTED
 (10/77) bufio:TestUnreadByteOthers: PASS (0.52 s)
 (09/77) bufio:TestUnreadByteMultiple: PASS (0.54 s)
 (12/77) bufio:TestUnreadRuneAtEOF: PASS (0.51 s)
 (11/77) bufio:TestUnreadRuneError: PASS (0.51 s)
 (13/77) bufio:TestReadWriteRune: PASS (0.51 s)
 (15/77) bufio:TestWriter: PASS (0.50 s)
 (14/77) bufio:TestReadStringAllocs: PASS (0.52 s)
 (16/77) bufio:TestWriteErrors: PASS (0.52 s)
 (17/77) bufio:TestNewReaderSizeIdempotent: STARTED
 (18/77) bufio:TestNewWriterSizeIdempotent: STARTED
 (19/77) bufio:TestWriteString: STARTED
 (20/77) bufio:TestBufferFull: STARTED
 (22/77) bufio:TestPeekThenUnreadRune: STARTED
 (23/77) bufio:TestReadLine: STARTED
 (21/77) bufio:TestPeek: STARTED
 (24/77) bufio:TestLineTooLong: STARTED
 (17/77) bufio:TestNewReaderSizeIdempotent: PASS (0.50 s)
 (19/77) bufio:TestWriteString: PASS (0.50 s)
 (18/77) bufio:TestNewWriterSizeIdempotent: PASS (0.51 s)
 (20/77) bufio:TestBufferFull: PASS (0.51 s)
 (22/77) bufio:TestPeekThenUnreadRune: PASS (0.50 s)
 (23/77) bufio:TestReadLine: PASS (0.51 s)
 (21/77) bufio:TestPeek: PASS (0.50 s)
 (24/77) bufio:TestLineTooLong: PASS (0.52 s)
 (25/77) bufio:TestReadAfterLines: STARTED
 (26/77) bufio:TestReadEmptyBuffer: STARTED
 (27/77) bufio:TestLinesAfterRead: STARTED
 (28/77) bufio:TestReadLineNonNilLineOrError: STARTED
 (29/77) bufio:TestReadLineNewlines: STARTED
 (31/77) bufio:TestReaderWriteToErrors: STARTED
 (30/77) bufio:TestReaderWriteTo: STARTED
 (32/77) bufio:TestWriterReadFrom: STARTED
 (32/77) bufio:TestWriterReadFrom: PASS (0.51 s)
 (25/77) bufio:TestReadAfterLines: PASS (1.01 s)
 (26/77) bufio:TestReadEmptyBuffer: PASS (1.01 s)
 (27/77) bufio:TestLinesAfterRead: PASS (1.01 s)
 (28/77) bufio:TestReadLineNonNilLineOrError: PASS (1.01 s)
 (29/77) bufio:TestReadLineNewlines: PASS (1.01 s)
 (31/77) bufio:TestReaderWriteToErrors: PASS (1.01 s)
 (30/77) bufio:TestReaderWriteTo: PASS (1.01 s)
 (33/77) bufio:TestWriterReadFromErrors: STARTED
 (34/77) bufio:TestWriterReadFromCounts: STARTED
 (35/77) bufio:TestNegativeRead: STARTED
 (36/77) bufio:TestReaderClearError: STARTED
 (37/77) bufio:TestWriterReadFromWhileFull: STARTED
 (39/77) bufio:TestWriterReadFromErrNoProgress: STARTED
 (38/77) bufio:TestWriterReadFromUntilEOF: STARTED
 (40/77) bufio:TestReadZero: STARTED
 (34/77) bufio:TestWriterReadFromCounts: PASS (0.51 s)
 (35/77) bufio:TestNegativeRead: PASS (0.51 s)
 (36/77) bufio:TestReaderClearError: PASS (0.50 s)
 (33/77) bufio:TestWriterReadFromErrors: PASS (1.01 s)
 (37/77) bufio:TestWriterReadFromWhileFull: PASS (0.51 s)
 (39/77) bufio:TestWriterReadFromErrNoProgress: PASS (0.51 s)
 (38/77) bufio:TestWriterReadFromUntilEOF: PASS (0.52 s)
 (40/77) bufio:TestReadZero: PASS (0.52 s)
 (41/77) bufio:TestReaderReset: STARTED
 (42/77) bufio:TestWriterReset: STARTED
 (43/77) bufio:TestReaderDiscard: STARTED
 (44/77) bufio:TestReaderSize: STARTED
 (45/77) bufio:TestWriterSize: STARTED
 (48/77) bufio:TestWriterReadFromMustReturnUnderlyingError: STARTED
 (46/77) bufio:TestPartialReadEOF: STARTED
 (47/77) bufio:TestWriterReadFromMustSetUnderlyingError: STARTED
 (41/77) bufio:TestReaderReset: PASS (0.51 s)
 (42/77) bufio:TestWriterReset: PASS (0.51 s)
 (45/77) bufio:TestWriterSize: PASS (0.50 s)
 (48/77) bufio:TestWriterReadFromMustReturnUnderlyingError: PASS (0.50 s)
 (46/77) bufio:TestPartialReadEOF: PASS (0.51 s)
 (47/77) bufio:TestWriterReadFromMustSetUnderlyingError: PASS (0.50 s)
 (50/77) bufio:ExampleScanner_lines: STARTED
 (49/77) bufio:ExampleWriter: STARTED
 (43/77) bufio:TestReaderDiscard: PASS (1.01 s)
 (44/77) bufio:TestReaderSize: PASS (1.01 s)
 (51/77) bufio:ExampleScanner_Bytes: STARTED
 (52/77) bufio:ExampleScanner_words: STARTED
 (53/77) bufio:ExampleScanner_custom: STARTED
 (54/77) bufio:ExampleScanner_emptyFinalToken: STARTED
 (49/77) bufio:ExampleWriter: PASS (0.51 s)
 (56/77) bufio:TestScanByte: STARTED
 (55/77) bufio:TestSpace: STARTED
 (51/77) bufio:ExampleScanner_Bytes: PASS (0.51 s)
 (52/77) bufio:ExampleScanner_words: PASS (0.52 s)
 (53/77) bufio:ExampleScanner_custom: PASS (0.51 s)
 (54/77) bufio:ExampleScanner_emptyFinalToken: PASS (0.52 s)
 (50/77) bufio:ExampleScanner_lines: PASS (1.00 s)
 (57/77) bufio:TestScanRune: STARTED
 (56/77) bufio:TestScanByte: PASS (0.50 s)
 (55/77) bufio:TestSpace: PASS (0.50 s)
 (58/77) bufio:TestScanWords: STARTED
 (59/77) bufio:TestScanLongLines: STARTED
 (60/77) bufio:TestScanLineTooLong: STARTED
 (57/77) bufio:TestScanRune: PASS (0.50 s)
 (62/77) bufio:TestScanLineReturnButNoNewline: STARTED
 (61/77) bufio:TestScanLineNoNewline: STARTED
 (64/77) bufio:TestScanLineEmptyFinalLineWithCR: STARTED
 (63/77) bufio:TestScanLineEmptyFinalLine: STARTED
 (59/77) bufio:TestScanLongLines: PASS (0.51 s)
 (60/77) bufio:TestScanLineTooLong: PASS (0.51 s)
 (62/77) bufio:TestScanLineReturnButNoNewline: PASS (0.50 s)
 (61/77) bufio:TestScanLineNoNewline: PASS (0.51 s)
 (65/77) bufio:TestSplitError: STARTED
 (58/77) bufio:TestScanWords: PASS (1.01 s)
 (64/77) bufio:TestScanLineEmptyFinalLineWithCR: PASS (0.50 s)
 (63/77) bufio:TestScanLineEmptyFinalLine: PASS (0.50 s)
 (67/77) bufio:TestNonEOFWithEmptyRead: STARTED
 (66/77) bufio:TestErrAtEOF: STARTED
 (68/77) bufio:TestBadReader: STARTED
 (65/77) bufio:TestSplitError: PASS (0.50 s)
 (69/77) bufio:TestScanWordsExcessiveWhiteSpace: STARTED
 (70/77) bufio:TestEmptyTokens: STARTED
 (71/77) bufio:TestWithNoEmptyTokens: STARTED
 (72/77) bufio:TestDontLoopForever: STARTED
 (67/77) bufio:TestNonEOFWithEmptyRead: PASS (0.51 s)
 (69/77) bufio:TestScanWordsExcessiveWhiteSpace: PASS (0.51 s)
 (73/77) bufio:TestBlankLines: STARTED
 (70/77) bufio:TestEmptyTokens: PASS (0.50 s)
 (71/77) bufio:TestWithNoEmptyTokens: PASS (0.51 s)
 (72/77) bufio:TestDontLoopForever: PASS (0.51 s)
 (66/77) bufio:TestErrAtEOF: PASS (1.01 s)
 (68/77) bufio:TestBadReader: PASS (1.02 s)
 (74/77) bufio:TestEmptyLinesOK: STARTED
 (75/77) bufio:TestHugeBuffer: STARTED
 (73/77) bufio:TestBlankLines: PASS (0.50 s)
 (76/77) bufio:TestNegativeEOFReader: STARTED
 (77/77) bufio:TestLargeReader: STARTED
 (74/77) bufio:TestEmptyLinesOK: PASS (0.50 s)
 (75/77) bufio:TestHugeBuffer: PASS (0.50 s)
 (76/77) bufio:TestNegativeEOFReader: PASS (0.50 s)
 (77/77) bufio:TestLargeReader: PASS (0.50 s)
RESULTS    : PASS 77 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2021-09-30T12.04-9feb393/results.html
JOB TIME   : 14.51 s
```
```

**Comment by ana** [comment]
Thanks for the update @clebergnu !

**Comment by ana** [comment]
I consider this fixed with 0a2c54128673b17b687981fa1e4c4188ed990914
